src/formatDate.js

```diff
@@ -9,6 +9,46 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const fromDelimeter = fromFormat[fromFormat.length - 1];
+  const dateArr = date.split(fromDelimeter);
+  const dataObj = {};
+
+  for (let d = 0; d <= 2; d++) {
+    dataObj[fromFormat[d]] = dateArr[d];
+  }
+
+  const properObj = {};
+  const propFormat = toFormat.slice(0, -1);
+
+  for (const d of propFormat) {
+    if (d === 'YY' || d === 'YYYY') {
+      properObj[d] = dataObj['YY'] || dataObj['YYYY'];
+    } else {
+      properObj[d] = dataObj[d];
+    }
+  }
```

Comment:
The current approach for populating `properObj` will not handle cases where the `fromFormat` and `toFormat` differ in the use of 'YY' and 'YYYY'. For example, if 'YYYY' is in `fromFormat` but 'YY' is in `toFormat`, it will not correctly shorten the year. Similarly, if 'YY' is in `fromFormat` but 'YYYY' is in `toFormat`, it will not correctly expand the year. You need to add logic to handle the conversion between 'YY' and 'YYYY' formats.